File: android/app/src/main/java/com/myvu/client/database/Reminder.java
package com.myvu.client.database;

public class Reminder {
    private final long id;
    private final String body;
    private final long triggerAt;
    private final String state;
    private final long createdAt;
    private final long updatedAt;
    private final Long firedAt;
    private final int snoozeCount;
    private final int alarmRequestCode;

    public Reminder(long id, String body, long triggerAt, String state, long createdAt, long updatedAt, Long firedAt, int snoozeCount, int alarmRequestCode) {
        this.id = id;
        this.body = body;
        this.triggerAt = triggerAt;
        this.state = state;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
        this.firedAt = firedAt;
        this.snoozeCount = snoozeCount;
        this.alarmRequestCode = alarmRequestCode;
    }

    public long getId() { return id; }
    public String getBody() { return body; }
    public long getTriggerAt() { return triggerAt; }
    public String getState() { return state; }
    public long getCreatedAt() { return createdAt; }
    public long getUpdatedAt() { return updatedAt; }
    public Long getFiredAt() { return firedAt; }
    public int getSnoozeCount() { return snoozeCount; }
    public int getAlarmRequestCode() { return alarmRequestCode; }
}

File: android-kotlin/app/src/main/java/com/myvu/client/database/Reminder.kt
package com.myvu.client.database

import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

data class Reminder(
    var id: Long = 0,
    var title: String = "",
    var body: String = "",
    var tags: String = "",
    var summary: String = "",
    var actionItems: String = "",
    var mindmapData: String = "",
    var attachmentsJson: String = "[]",
    var triggerAt: Long = 0,
    var createdAt: Long = System.currentTimeMillis(),
    var updatedAt: Long = System.currentTimeMillis(),
    var state: String = "PENDING", // PENDING, COMPLETED, SNOOZED, CANCELLED
    var alarmRequestCode: Int = 0
) {
    val tagsList: List<String>
        get() = if (tags.isBlank()) emptyList() else tags.split(",").map { it.trim() }.filter { it.isNotEmpty() }

    fun getAttachments(): List<Attachment> = Attachment.listFromJson(attachmentsJson)

    fun formattedTriggerDate(): String {
        val sdf = SimpleDateFormat("dd MMM, HH:mm", Locale.getDefault())
        return sdf.format(Date(triggerAt))
    }

    fun formattedDate(): String {
        val time = if (createdAt > 0) createdAt else System.currentTimeMillis()
        val sdf = SimpleDateFormat("dd MMM, HH:mm", Locale.getDefault())
        return sdf.format(Date(time))
    }
}
